src/background/service-worker-init.ts

```diff
@@ -1,20 +1,230 @@
+import { TelemetryEventHandler } from 'background/telemetry/telemetry-event-handler';
+import { TelemetryLogger } from 'background/telemetry/telemetry-logger';
+import { TelemetryStateListener } from 'background/telemetry/telemetry-state-listener';
+import { UsageLogger } from 'background/usage-logger';
+import { createToolData } from 'common/application-properties-provider';
+import { AxeInfo } from 'common/axe-info';
+import { BrowserAdapterFactory } from 'common/browser-adapters/browser-adapter-factory';
+import { WebVisualizationConfigurationFactory } from 'common/configs/web-visualization-configuration-factory';
+import { DateProvider } from 'common/date-provider';
+import { getIndexedDBStore } from 'common/indexedDB/get-indexeddb-store';
+import { IndexedDBAPI, IndexedDBUtil } from 'common/indexedDB/indexedDB';
+import { createDefaultLogger } from 'common/logging/default-logger';
+import { NavigatorUtils } from 'common/navigator-utils';
+import { NotificationCreator } from 'common/notification-creator';
+import { createDefaultPromiseFactory } from 'common/promises/promise-factory';
+import { TelemetryDataFactory } from 'common/telemetry-data-factory';
+import { UrlValidator } from 'common/url-validator';
+import { WindowUtils } from 'common/window-utils';
+import { title, toolName } from 'content/strings/application';
+import { IssueFilingServiceProviderImpl } from 'issue-filing/issue-filing-service-provider-impl';
+import UAParser from 'ua-parser-js';
+import { deprecatedStorageDataKeys, storageDataKeys } from './local-storage-data-keys';
+import { cleanKeysFromStorage } from './user-stored-data-cleaner';
 
-const state = {
-    count: 0,
-    interval: 0,
-};
+async function initialize(): Promise<void> {
+    const userAgentParser = new UAParser(globalThis.navigator.userAgent);
+    const browserAdapterFactory = new BrowserAdapterFactory(userAgentParser);
+    const browserAdapter = browserAdapterFactory.makeFromUserAgent();
 
-chrome.action.onClicked.addListener(onClicked);
-chrome.runtime.onInstalled.addListener(onInstalled);
+    // This only removes keys that are unused by current versions of the extension, so it's okay for it to race with everything else
+    const cleanKeysFromStoragePromise = cleanKeysFromStorage(
+        browserAdapter,
+        deprecatedStorageDataKeys,
+    );
 
-function onInstalled() {
-    state.interval = Math.ceil(5 * Math.random());
-    console.log(`interval = ${state.interval}`);
-}
+    const urlValidator = new UrlValidator(browserAdapter);
+    const indexedDBInstance: IndexedDBAPI = new IndexedDBUtil(getIndexedDBStore());
+    const indexedDBDataKeysToFetch = [
+        IndexedDBDataKeys.assessmentStore,
+        IndexedDBDataKeys.userConfiguration,
+    ];
+
+    // // These can run concurrently, both because they are read-only and because they use different types of underlying storage
+    const persistedDataPromise = getPersistedData(indexedDBInstance, indexedDBDataKeysToFetch);
```

Comment:
My changes in https://github.com/microsoft/accessibility-insights-web/pull/5389 will conflict with this/ change some of this logic, so we should probably try to get this PR in before merging that one.